### ISSUE TITLE
[Quant] Validate FixedQParams observers in eager mode

### ISSUE THREAD

**Issue by andrewor14** [pytorch/pytorch#88579]
Currently, in eager mode, we treat fixed qparams ops such as `torch.nn.Sigmoid` and `torch.nn.Tanh` as having "special activation post process" according to this mapping:
https://github.com/pytorch/pytorch/blob/a16ced03c93dcbc5b08d0f9a36f8feab583f129a/torch/ao/quantization/quantization_mappings.py#L188-L193

Even if the user explicitly configured a QConfig for these modules, we would overwrite the configured observers with a `FixedQParamsFakeQuantize`. https://github.com/pytorch/pytorch/pull/88498 recently added a workaround for users to override this behavior, but requires users to explicitly specify these modules in `non_leaf_module_list`.

Instead, what we should do is don't overwrite the observers set by the users, but instead validate whether these observers satisfy our requirements. We did this recently with FX graph mode quantization already (https://github.com/pytorch/pytorch/pull/80184) and we should do the same for eager mode.

**Comment by andrewor14** [comment]
cc @jerryzh168
